lldb/tools/lldb-dap/tool/lldb-dap.cpp

```diff
@@ -226,23 +226,32 @@ static llvm::Expected<std::pair<Socket::SocketProtocol, std::string>>
 validateConnection(llvm::StringRef conn) {
   auto uri = lldb_private::URI::Parse(conn);
 
-  if (uri && (uri->scheme == "tcp" || uri->scheme == "connect" ||
-              !uri->hostname.empty() || uri->port)) {
+  auto make_error = [conn]() -> llvm::Error {
```

Comment:
I think we also need to update line 130 from `lldb-dap --connection connection://localhost:<port>` to `lldb-dap --connection listen://[host]:<port>`